src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessor.java

```diff
@@ -126,21 +134,31 @@ private void finaliseProcessedMessage(
                 log.info("Message with ID {} has been completed", message.getMessageId());
                 break;
             case UNRECOVERABLE_FAILURE:
-                messageOperations.deadLetter(
-                    message.getLockToken(),
+                deadLetterTheMessage(
+                    message,
                     "Message processing error",
                     processingResult.exception.getMessage()
                 );
 
-                log.info("Message with ID {} has been dead-lettered", message.getMessageId());
                 break;
             case POTENTIALLY_RECOVERABLE_FAILURE:
-                // do nothing - let the message lock expire
-                log.info(
-                    "Allowing message with ID {} to return to queue (delivery attempt {})",
-                    message.getMessageId(),
-                    message.getDeliveryCount() + 1
-                );
+                int deliveryCount = (int) message.getDeliveryCount() + 1;
+
+                if (deliveryCount < maxDeliveryCount - 5) {
```

Comment:
looks like actual max is not `maxDeliveryCount`

Reply:
This idea was proposed initially: increase delivery count and dead letter prior that. For fairly low value like `5` application application makes sure it gets to the point. Would not like to trust azure service bus for handling situation "whether I did push X times yet or not" since 1st delivery is marked `0`. Even though we know and can inspect such behaviour in our logs.

Consider it as safety measure. Default configuration is 10 deliveries so it'll die out sooner. The bigger the count - the more attempts. Hence the increase which is linked in the description.

If we're happy with `deliveryCount < maxDeliveryCount` instead - let's agree in conversation among the team ;)